viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+    """ user_data dict with watched list of movies dictionaries and friends
+    user_data = {
+        watched[movies{title:}],
+        friends[friend{watched[movies{title:}]}]                  
+    }
+    """
+    """
+movies that user has watched
+movies that friend has watched
+which movies the user has watched but none friends watched  movies user watch != movies friends watched 
+return unique[movies{}]
+    """
+    unique_movies = []
+
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for movie in movie_list:
+        title = movie["title"]
+        unique = True
+        for friend in friends_list:
+            friend_movies_list = friend["watched"]
+            for friend_movie in friend_movies_list:
+                if friend_movie["title"] == title:
+                    unique = False
+                    break
+            if not unique:
+                break
+        if unique:
+            unique_movies.append(movie)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+def get_available_recs(user_data):
+    """user_data
+    {    subscriptions["name, name , name"]}
+    """
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies and friend_movie["host"] in user_data["subscriptions"]:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+
+def get_new_rec_by_genre(user_data):
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    genre = get_most_watched_genre(user_data)
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
```

Comment:
Similar to the previous function, notice that this code is very similar to some of our previous functions. Could we simplify this by making use of a function that gives us the unique movies watched by the user's friends?